projects/chaos/src/bps_heartbeat.c

```diff
@@ -0,0 +1,53 @@
+#include "bps_heartbeat.h"
```

Comment:
Comment?

Reply:
Done. Only general comments in `*.h` since implementation is much simpler than powertrain.